### PR TITLE
Do not create aggregated ByteBuf when publisher empty or first element empty

### PR DESCRIPTION
**Pull request by violetagg**
The PR is scheduled for `0.9`

### CODE REVIEW

File: src/main/java/reactor/netty/http/client/HttpClientOperations.java [hunk 1]
```diff
@@ -796,16 +796,28 @@ public NettyOutbound sendObject(Object message) {
...
+			                   if (buf != null && buf.readableBytes() > 0) {
+			                       return flux.collect(alloc::heapBuffer, ByteBuf::writeBytes)
+			                                  .flatMap(agg -> {
+			                                      if (!HttpUtil.isTransferEncodingChunked(request) &&
+			                                              !HttpUtil.isContentLengthSet(request)) {
+			                                          request.headers()
+			                                                 .setInt(HttpHeaderNames.CONTENT_LENGTH, agg.readableBytes());
+			                                      }
+			                                      return parent.then()
+			                                                   .thenEmpty(FutureMono.disposableWriteAndFlush(parent.channel(), Mono.just(agg)));
```
Comment (bsideup) [inline src/main/java/reactor/netty/http/client/HttpClientOperations.java]: I think you can drop `.then()` here

File: src/main/java/reactor/netty/http/client/HttpClientOperations.java [hunk 2]
```diff
@@ -796,16 +796,28 @@ public NettyOutbound sendObject(Object message) {
...
+			                                  });
+			                   }
+			               }
+			               return parent.then();
```
Comment (bsideup) [inline src/main/java/reactor/netty/http/client/HttpClientOperations.java]: `.then()` here is redundant, there is `.then` after `switchOnFirst` (or maybe that `.then` can be dropped)

File: src/main/java/reactor/netty/http/client/HttpClientOperations.java (hunk 1 above)
Comment (violetagg) [reply src/main/java/reactor/netty/http/client/HttpClientOperations.java]: `.then()` cannot be dropped - on this step Reactor Netty sends the headers for the outbound message.

File: src/main/java/reactor/netty/http/client/HttpClientOperations.java (hunk 2 above)
Comment (violetagg) [reply src/main/java/reactor/netty/http/client/HttpClientOperations.java]: `.then()` cannot be dropped - on this step Reactor Netty sends the headers for the outbound message.

**Pull request merged**